### PR TITLE
PEP 3106 -- Revamping dict.keys(), .values() and .items()

### PR DESCRIPTION
**Pull request by simplicbe**
Implementing new dictionary iterator and view functions: https://www.python.org/dev/peps/pep-3135/

Task: https://github.com/IronLanguages/ironpython3/issues/13

### CODE REVIEW

**Comment by simplicbe** [comment]
Anything here? If not, I merge this.

File: Src/IronPython/Runtime/Types/DictProxy.cs [hunk 1]
```diff
@@ -77,17 +77,18 @@ public class DictProxy : IDictionary, IEnumerable, IDictionary<object, object> {
...
-        }
-
-        public IEnumerator iterkeys(CodeContext/*!*/ context) {
-            return new DictionaryKeyEnumerator(_dt.GetMemberDictionary(context, false)._storage);
-        }
-
-        public IEnumerator itervalues(CodeContext/*!*/ context) {
-            return new DictionaryValueEnumerator(_dt.GetMemberDictionary(context, false)._storage);
-        }
+        //TODO: DICT_ITER
```
Comment (simplicbe) [inline Src/IronPython/Runtime/Types/DictProxy.cs]: Oh, I will clean up this.

**Comment by slide** [comment]
Is there a test for this?

File: Src/IronPython/Runtime/DictionaryOps.cs [hunk 2]
```diff
@@ -93,16 +93,27 @@ public static class DictionaryOps {
             return ret;
         }
 
-        public static IEnumerator iteritems(IDictionary<object, object> self) {
-            return ((IEnumerable)items(self)).GetEnumerator();
+        public static List keys(IDictionary<object, object> self)
```
Comment (slide) [inline Src/IronPython/Runtime/DictionaryOps.cs]: This isn't quite correct. keys() returns a dict_keys object in CPython.

Comment (slide) [inline Src/IronPython/Runtime/DictionaryOps.cs]:
```
>>> type(d.keys())
<class 'dict_keys'>
>>> type(d.keys()).__bases__
(<class 'object'>,)
```

**Comment by slide** [comment]
This needs to be revamped quite a bit based on the dict_keys and dict_values views.

**Pull request closed without merging**